keygen/src/json_model.h

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2020 Xaptum, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef _JSON_MODEL_H
+#define _JSON_MODEL_H
+
+#include <jansson.h>
+
+struct xcr_login {
```

Comment:
It might be helpful for internal consistency to mimic the naming scheme from the [Go SDK](https://github.com/xaptum/go-enf/blob/master/enf/auth.go).

So 
- `xcr.{h,c}` would become `auth.{h,c}`.
- `xcr_login` would become `auth_request`
- `xcr_auth_data` would become `credentials`
- `xcr_page` would become `auth_page`.


The `xiam` functions aren't in the `go-enf` API yet (or in `enfun-keygen`!) , but those would get names like
- `iam.{h.c}`
- `iam_create_endpoint_request` (a `NEW_ENDPOINT_AUTH` object)
- `iam_endpoint` (the returned `ENDPOINT` object)